satpy/readers/sar_c_safe.py

```diff
@@ -309,125 +313,65 @@ def __init__(self, filename, filename_info, filetype_info, calfh, noisefh):
 
     def get_dataset(self, key, info):
         """Load a dataset."""
-        if self._polarization != key.polarization:
+        if self._polarization != key["polarization"]:
             return
 
         logger.debug('Reading %s.', key['name'])
 
-        if key['name'] in ['longitude', 'latitude']:
+        if key['name'] in ['longitude', 'latitude', 'altitude']:
             logger.debug('Constructing coordinate arrays.')
 
-            if self.lons is None or self.lats is None:
-                self.lons, self.lats, self.alts = self.get_lonlatalts()
+            lons, lats, alts = self.get_lonlatalts()
 
             if key['name'] == 'latitude':
-                data = self.lats
-            else:
-                data = self.lons
+                data = lats
+            elif key['name'] == 'longitude':
+                data = lons
+            elif key['name'] == 'altitude':
+                data = alts
```

Comment:
This set of if statements screams "dictionary" or something else more elegant, but since it is only 3 options maybe it is fine.

Reply:
Good point, fixed.

Reply:
I'm not sure this solution is any better...but it is different :laughing: Could be a separate method even, but not required to change it.

Fine by me.

Reply:
What did you have in mind?

Reply:
I honestly don't know. That's why I said "maybe it is fine". This is the kind of code that if I wrote it myself I would spend way too long trying to make it look better because it "feels" like it should be easier/cleaner, but what that ends up looking like I don't know. Of the top of my head:

```
coord_indexes = {'longitude': 0, 'latitude': 1, 'altitude': 2}
data = self.get_lonlatalts()[coord_indexes[key['name']]]
```

:man_shrugging: I don't know. Just leave it.